libethcore/TransactionBase.h

```diff
@@ -168,7 +167,9 @@ class TransactionBase
     u256 m_gas;							///< The total gas to convert, paid for from sender's account. Any unused gas gets refunded once the contract is ended.
     bytes m_data;						///< The data associated with the transaction, or the initialiser if it's a creation transaction.
     boost::optional<SignatureStruct> m_vrs;	///< The signature of the transaction. Encodes the sender.
-    int m_chainId = -4;					///< EIP155 value for calculating transaction hash https://github.com/ethereum/EIPs/issues/155
+    /// EIP155 value for calculating transaction hash
+    /// https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+    boost::optional<uint64_t> m_chainId;
```

Comment:
Should we use std::optional now that we can build C++17 on Windows? Or should we just convert all boost::optional to std::optional in one go at some point in the future?

Reply:
We still build with C++11 standard at the moment (it should be a separate PR when we want to switch; I'm not sure whether AppVeyor supports it)

Reply:
Should be fine in VS 2017.